OpenRA.Game/Graphics/SpriteCache.cs

```diff
@@ -93,9 +93,11 @@ public void LoadReservations(ModData modData)
 						{
 							var resolved = new Sprite[loadedFrames.Length];
 							resolvedSprites[token] = resolved;
-							var frames = rs.Frames ?? Enumerable.Range(0, loadedFrames.Length);
+							if (rs.Frames != null && rs.Frames.Any(i => i >= loadedFrames.Length))
+								throw new InvalidOperationException($"{rs.Location}: {filename} does not contain frames: " +
+									$"{string.Join(',', rs.Frames.Where(f => f >= loadedFrames.Length))}");
```

Comment:
leave out the `$"{}"` - if it's on two seperate lines.

Reply:
fixed